### ISSUE TITLE
Renaming sometimes renames also the path

### ISSUE THREAD

**Issue by erikari** [EsotericSoftware/spine-editor#227]
**Expected behavior:**
In "Find and Replace" if I replace the "Name" field of an attachment but I don't select the path, I expect it to be renamed, but not the path.

**Results:**
I bulk rename attachments after importing them in a preexisting slot, and the attachments go missing.

**Repro steps:**
1. open spineboy
2. select the eye slot and expand it, then select the two attachments inside
3. open Find and Replace
4. in Find write "eye", Replace "moo-is-cute", 
Options: Match Case, First occurence are selected, RegEX is not.
Scope: Tree Selection
Field: Name
Types: Region
5. click replace
!(https://image.prntscr.com/image/9ufxBQ_QQlWJTL8xNap4QA.png)
Image goes missing.
!(https://image.prntscr.com/image/P-F8M27CR26BF3zSez-E1w.png)

Basically it happens with anything you decide to replace, even if you select both the slot and the attachment, or many slots and attachments, as long as the attachment is modified of course.

I'm not 100% sure this is a bug rather than an expected behavior. 
Also it was not happening on some of my attachments because I had previously modified their path and only later changed their names to whatever I liked. 

Sounds a bit confusing that first the both are linked and and only after modifying the path they aren't anymore.

**Comment by NathanSweet** [comment]
The spineboy eye images have a blank path. Finding images uses the name if the path is blank. When you change the name (eg using Find and Replace or just by renaming the attachment), the path is still blank, so it uses the new name to find the image, which fails.

What is the behavior you would like? You want to rename the attachment and move the old name to the path? I don't think we'd want it to always do that, many times Find and Replace is used to fix up names after moving the images around on disk. You want a checkbox on the Find and Replace dialog? What about the normal attachment rename dialog? The same behavior occurs there.

**Comment by NathanSweet** [comment]
@erikari  No response!

**Comment by erikari** [comment]
My bad, I was looking at this a couple days ago and thinking about it, but renaming as it is, is ultimately fine.  Changing this would need to add two fields, the name field (the name inside spine) and the path field, and it's not worth it /:

**Comment by NathanSweet** [comment]
I wasn't sure what exactly was being proposed. If you like, you can propose something specific if it would solve your problem.

**Comment by erikari** [comment]
This was related to a  system of importing alt frames I was using:
first we'd call all the attachments in photoshop like this:
- leg-attack
- leg-idle
- leg-jump
- arm-attack
- arm-jump

so when importing them in spine, the image folder would list them in a way so that they'd be easy to add to the right slot massively.

then I'd change their name just for spine (with lovely regex) to invert the first name with the second, so that filtering the tree by attachment would show the right attachments for the animation to activate (therefore useful for the animator):
- attack-arm
- attack-leg
- jump-arm
- jump-leg
- idle-leg

but the path would have to keep the original name in order for this trick to work, therefore the new names would be used just in spine.

**Comment by erikari** [comment]
But if three filtering will be implemented any time soon, this system won't be needed anymore u.u
